File: binding/include/charts/libqhboxplotmodelmapper.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");%%_IMPORTLIBS_%%
%%_STRUCTDEFS_%%

/// https://doc.qt.io/qt-6/qhboxplotmodelmapper.html
pub const qhboxplotmodelmapper = struct {

    /// New constructs a new QHBoxPlotModelMapper object.
    pub fn New() QtC.QHBoxPlotModelMapper {
        return qtc.QHBoxPlotModelMapper_new();
    }


    /// New2 constructs a new QHBoxPlotModelMapper object.
    pub fn New2(parent: ?*anyopaque) QtC.QHBoxPlotModelMapper {
        return qtc.QHBoxPlotModelMapper_new2(@ptrCast(parent));
    }


    pub fn Tr(s: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QHBoxPlotModelMapper_Tr(@ptrCast(s));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qhboxplotmodelmapper.Tr: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn Model(self: ?*anyopaque, ) QtC.QAbstractItemModel {
        return qtc.QHBoxPlotModelMapper_Model(@ptrCast(self));
    }

    pub fn SetModel(self: ?*anyopaque, model: ?*anyopaque) void {
        qtc.QHBoxPlotModelMapper_SetModel(@ptrCast(self), @ptrCast(model));
    }

    pub fn Series(self: ?*anyopaque, ) QtC.QBoxPlotSeries {
        return qtc.QHBoxPlotModelMapper_Series(@ptrCast(self));
    }

    pub fn SetSeries(self: ?*anyopaque, series: ?*anyopaque) void {
        qtc.QHBoxPlotModelMapper_SetSeries(@ptrCast(self), @ptrCast(series));
    }

    pub fn FirstBoxSetRow(self: ?*anyopaque, ) i32 {
        return qtc.QHBoxPlotModelMapper_FirstBoxSetRow(@ptrCast(self));
    }

    pub fn SetFirstBoxSetRow(self: ?*anyopaque, firstBoxSetRow: i32) void {
        qtc.QHBoxPlotModelMapper_SetFirstBoxSetRow(@ptrCast(self), firstBoxSetRow);
    }

    pub fn LastBoxSetRow(self: ?*anyopaque, ) i32 {
        return qtc.QHBoxPlotModelMapper_LastBoxSetRow(@ptrCast(self));
    }

    pub fn SetLastBoxSetRow(self: ?*anyopaque, lastBoxSetRow: i32) void {
        qtc.QHBoxPlotModelMapper_SetLastBoxSetRow(@ptrCast(self), lastBoxSetRow);
    }

    pub fn FirstColumn(self: ?*anyopaque, ) i32 {
        return qtc.QHBoxPlotModelMapper_FirstColumn(@ptrCast(self));
    }

    pub fn SetFirstColumn(self: ?*anyopaque, firstColumn: i32) void {
        qtc.QHBoxPlotModelMapper_SetFirstColumn(@ptrCast(self), firstColumn);
    }

    pub fn ColumnCount(self: ?*anyopaque, ) i32 {
        return qtc.QHBoxPlotModelMapper_ColumnCount(@ptrCast(self));
    }

    pub fn SetColumnCount(self: ?*anyopaque, rowCount: i32) void {
        qtc.QHBoxPlotModelMapper_SetColumnCount(@ptrCast(self), rowCount);
    }

    pub fn SeriesReplaced(self: ?*anyopaque, ) void {
        qtc.QHBoxPlotModelMapper_SeriesReplaced(@ptrCast(self));
    }

    pub fn OnSeriesReplaced(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QHBoxPlotModelMapper_Connect_SeriesReplaced(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn ModelReplaced(self: ?*anyopaque, ) void {
        qtc.QHBoxPlotModelMapper_ModelReplaced(@ptrCast(self));
    }

    pub fn OnModelReplaced(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QHBoxPlotModelMapper_Connect_ModelReplaced(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn FirstBoxSetRowChanged(self: ?*anyopaque, ) void {
        qtc.QHBoxPlotModelMapper_FirstBoxSetRowChanged(@ptrCast(self));
    }

    pub fn OnFirstBoxSetRowChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QHBoxPlotModelMapper_Connect_FirstBoxSetRowChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn LastBoxSetRowChanged(self: ?*anyopaque, ) void {
        qtc.QHBoxPlotModelMapper_LastBoxSetRowChanged(@ptrCast(self));
    }

    pub fn OnLastBoxSetRowChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QHBoxPlotModelMapper_Connect_LastBoxSetRowChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn FirstColumnChanged(self: ?*anyopaque, ) void {
        qtc.QHBoxPlotModelMapper_FirstColumnChanged(@ptrCast(self));
    }

    pub fn OnFirstColumnChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QHBoxPlotModelMapper_Connect_FirstColumnChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn ColumnCountChanged(self: ?*anyopaque, ) void {
        qtc.QHBoxPlotModelMapper_ColumnCountChanged(@ptrCast(self));
    }

    pub fn OnColumnCountChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QHBoxPlotModelMapper_Connect_ColumnCountChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn Tr2(s: []const u8, c: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QHBoxPlotModelMapper_Tr2(@ptrCast(s), @ptrCast(c));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qhboxplotmodelmapper.Tr2: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn Tr3(s: []const u8, c: []const u8, n: i32, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QHBoxPlotModelMapper_Tr3(@ptrCast(s), @ptrCast(c), n);
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qhboxplotmodelmapper.Tr3: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QHBoxPlotModelMapper_Delete(@ptrCast(self));
    }
};

File: binding/src/charts/libqhboxplotmodelmapper.cpp
#include <QAbstractItemModel>
#include <QBoxPlotModelMapper>
#include <QBoxPlotSeries>
#include <QHBoxPlotModelMapper>
#include <QObject>
#include <QString>
#include <QByteArray>
#include <cstring>
#include <type_traits>
#include <qhboxplotmodelmapper.h>
#include "libqhboxplotmodelmapper.h"
#include "libqhboxplotmodelmapper.hxx"

QHBoxPlotModelMapper* QHBoxPlotModelMapper_new() {
	 return new QHBoxPlotModelMapper();
}

QHBoxPlotModelMapper* QHBoxPlotModelMapper_new2(QObject* parent) {
	 return new QHBoxPlotModelMapper(parent);
}

libqt_string QHBoxPlotModelMapper_Tr(const char* s) {
	QString _ret = QHBoxPlotModelMapper::tr(s);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

QAbstractItemModel* QHBoxPlotModelMapper_Model(const QHBoxPlotModelMapper* self) {
	return self->model();
}

void QHBoxPlotModelMapper_SetModel(QHBoxPlotModelMapper* self, QAbstractItemModel* model) {
	self->setModel(model);
}

QBoxPlotSeries* QHBoxPlotModelMapper_Series(const QHBoxPlotModelMapper* self) {
	return self->series();
}

void QHBoxPlotModelMapper_SetSeries(QHBoxPlotModelMapper* self, QBoxPlotSeries* series) {
	self->setSeries(series);
}

int QHBoxPlotModelMapper_FirstBoxSetRow(const QHBoxPlotModelMapper* self) {
	return self->firstBoxSetRow();
}

void QHBoxPlotModelMapper_SetFirstBoxSetRow(QHBoxPlotModelMapper* self, int firstBoxSetRow) {
	self->setFirstBoxSetRow(firstBoxSetRow);
}

int QHBoxPlotModelMapper_LastBoxSetRow(const QHBoxPlotModelMapper* self) {
	return self->lastBoxSetRow();
}

void QHBoxPlotModelMapper_SetLastBoxSetRow(QHBoxPlotModelMapper* self, int lastBoxSetRow) {
	self->setLastBoxSetRow(lastBoxSetRow);
}

int QHBoxPlotModelMapper_FirstColumn(const QHBoxPlotModelMapper* self) {
	return self->firstColumn();
}

void QHBoxPlotModelMapper_SetFirstColumn(QHBoxPlotModelMapper* self, int firstColumn) {
	self->setFirstColumn(firstColumn);
}

int QHBoxPlotModelMapper_ColumnCount(const QHBoxPlotModelMapper* self) {
	return self->columnCount();
}

void QHBoxPlotModelMapper_SetColumnCount(QHBoxPlotModelMapper* self, int rowCount) {
	self->setColumnCount(rowCount);
}

void QHBoxPlotModelMapper_SeriesReplaced(QHBoxPlotModelMapper* self) {
	self->seriesReplaced();
}

void QHBoxPlotModelMapper_Connect_SeriesReplaced(QHBoxPlotModelMapper* self, intptr_t slot) {
    void (*slotFunc)(QHBoxPlotModelMapper*) = reinterpret_cast<void (*)(QHBoxPlotModelMapper*)>(slot);
    QHBoxPlotModelMapper::connect(self, &QHBoxPlotModelMapper::seriesReplaced, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QHBoxPlotModelMapper_ModelReplaced(QHBoxPlotModelMapper* self) {
	self->modelReplaced();
}

void QHBoxPlotModelMapper_Connect_ModelReplaced(QHBoxPlotModelMapper* self, intptr_t slot) {
    void (*slotFunc)(QHBoxPlotModelMapper*) = reinterpret_cast<void (*)(QHBoxPlotModelMapper*)>(slot);
    QHBoxPlotModelMapper::connect(self, &QHBoxPlotModelMapper::modelReplaced, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QHBoxPlotModelMapper_FirstBoxSetRowChanged(QHBoxPlotModelMapper* self) {
	self->firstBoxSetRowChanged();
}

void QHBoxPlotModelMapper_Connect_FirstBoxSetRowChanged(QHBoxPlotModelMapper* self, intptr_t slot) {
    void (*slotFunc)(QHBoxPlotModelMapper*) = reinterpret_cast<void (*)(QHBoxPlotModelMapper*)>(slot);
    QHBoxPlotModelMapper::connect(self, &QHBoxPlotModelMapper::firstBoxSetRowChanged, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QHBoxPlotModelMapper_LastBoxSetRowChanged(QHBoxPlotModelMapper* self) {
	self->lastBoxSetRowChanged();
}

void QHBoxPlotModelMapper_Connect_LastBoxSetRowChanged(QHBoxPlotModelMapper* self, intptr_t slot) {
    void (*slotFunc)(QHBoxPlotModelMapper*) = reinterpret_cast<void (*)(QHBoxPlotModelMapper*)>(slot);
    QHBoxPlotModelMapper::connect(self, &QHBoxPlotModelMapper::lastBoxSetRowChanged, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QHBoxPlotModelMapper_FirstColumnChanged(QHBoxPlotModelMapper* self) {
	self->firstColumnChanged();
}

void QHBoxPlotModelMapper_Connect_FirstColumnChanged(QHBoxPlotModelMapper* self, intptr_t slot) {
    void (*slotFunc)(QHBoxPlotModelMapper*) = reinterpret_cast<void (*)(QHBoxPlotModelMapper*)>(slot);
    QHBoxPlotModelMapper::connect(self, &QHBoxPlotModelMapper::firstColumnChanged, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QHBoxPlotModelMapper_ColumnCountChanged(QHBoxPlotModelMapper* self) {
	self->columnCountChanged();
}

void QHBoxPlotModelMapper_Connect_ColumnCountChanged(QHBoxPlotModelMapper* self, intptr_t slot) {
    void (*slotFunc)(QHBoxPlotModelMapper*) = reinterpret_cast<void (*)(QHBoxPlotModelMapper*)>(slot);
    QHBoxPlotModelMapper::connect(self, &QHBoxPlotModelMapper::columnCountChanged, [self, slotFunc]() {
	slotFunc(self);
    });
}

libqt_string QHBoxPlotModelMapper_Tr2(const char* s, const char* c) {
	QString _ret = QHBoxPlotModelMapper::tr(s, c);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

libqt_string QHBoxPlotModelMapper_Tr3(const char* s, const char* c, int n) {
	QString _ret = QHBoxPlotModelMapper::tr(s, c, n);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

void QHBoxPlotModelMapper_Delete(QHBoxPlotModelMapper* self) {
    delete self;
}

